scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py

```diff
@@ -109,10 +109,20 @@ def __init__(self, parent=None):
         self.electron_peaks = self._get_electron_peaks()
         self._generate_element_widgets()
 
-    # Return an unused colour, if all used then start with the first one
-    def get_color(self):
-        color = "C%d" % self.color_index
-        self.color_index = (self.color_index + 1) % self.num_colors
+    # Return the first unused colour, if all used then restart from the beginning of the cycle
+    def get_color(self, element):
+        if element in self.used_colors:
+            return self.used_colors[element]
+
+        occurrences = []
```

Comment:
```suggestion
        occurrences = [self.used_colors.values().count('C{}'.format(i)) for i in range(self.num_colors)]
```
This seems clearer to be, but maybe I've done too much Python recently. If you choose to go this way this makes the 2 lines underneath unnecessary